kubespawner/reflector.py

```diff
@@ -132,6 +155,16 @@ def _watch_and_update(self):
         update' cycle on them), we should be ok!
         """
         cur_delay = 0.1
+        watch_args = {
+            'namespace': self.namespace,
+            'label_selector': self.label_selector,
+        }
+        if self.request_timeout:
+            watch_args['_request_timeout'] = self.request_timeout
```

Comment:
What do we lose by not using this private argument? Will it never timeout, or will it just not be configurable by us? If it's not configurable, I'd rather respect the private API and leave it out.

Reply:
This private `_request_timeout` is about the underlying urllib3 timeout. I have not found another way to set that. Could probably be raised as an issues with the swagger code generator.

Reply:
here is the "documentation" I have found. https://github.com/kubernetes-client/python/blob/master/kubernetes/client/rest.py#L125